keras/engine/training.py

```diff
@@ -2219,8 +2235,12 @@ def generate_arrays_from_file(path):
                     break
 
         finally:
-            if enqueuer is not None:
-                enqueuer.stop()
+            try:
```

Comment:
Why two nested try catch ?

Reply:
If an exception is raised in ```enqueuer.stop```, we would not stop the ```val_enqueuer```.
I guess alternatively one could make the enqueuers context managers.